mcrouter/lib/network/ThreadLocalSSLContextProvider.cpp

```diff
@@ -308,7 +308,7 @@ std::shared_ptr<SSLContext> createClientSSLContext(
   auto context = std::make_shared<ClientSSLContext>(ticketCache.get());
   auto ciphers = folly::ssl::SSLCommonOptions::ciphers();
   std::vector<std::string> cVec(ciphers.begin(), ciphers.end());
-#if FOLLY_OPENSSL_HAS_ALPN
+#if FOLLY_OPENSSL_IS_110
```

Comment:
ALPN was introduced prior to the security level setting, so changing this entire block (modifying the cipher list, then setting the ALPN) is too restrictive. (ALPN is in 1.0.2, security level related items were introduced in openssl/openssl@b362ccab5c1d52086f19d29a32f4acc11073b86b which was in 1.1.0)

The preprocessor guard should probably be only for the `SSL_CTX_set_security_level` call.

Reply:
Thanks for taking a look. I've updated to only wrap `SSL_CTX_set_security_level` with the preprocessor guard.